src/api/routes/company.js

```diff
@@ -72,6 +72,15 @@ export default (app) => {
 
     });
 
+    router.get("/:companyId/profile", validators.profile, async (req, res) => {
+        const company = await new CompanyService().findById(req.params.companyId, req.hasAdminPrivileges, req.hasAdminPrivileges);
+        const offers = (await new OfferService()
+            .getOffersByCompanyId(req.params.companyId, req.targetOwner, req.hasAdminPrivileges))
+            .sort((a1, a2) => a1.publishDate > a2.publishDate)
+            .slice(0, 5);
```

Comment:
You should sort and limit on the query, not after getting stuff from DB

Reply:
I tried looking into doing this, however, `getOffersByCompanyId` currently does not return the mongoose query, and therefore I'm unable to use Mongoose to sort and limit it. I kept it as it was in this last commit, as I'm unsure if I should change the function or just keep it as it is.

Reply:
Another solution could be to refactor the current method so that it returns a query instead of the results (correct me if I'm wrong but most of, if not all, service methods return queries), this way we would only have the "base" code written once (fetching offers based on the company that owns them) and then the route itself would do more processing in order to get the desired result. What do you think ?

Reply:
I agree it'd be best, if possible